controller/client/monitor_client.go

```diff
@@ -23,7 +23,6 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 	"k8s.io/apimachinery/pkg/runtime"
-	"sigs.k8s.io/controller-runtime/pkg/cache"
```

Comment:
Why do we remove it? is it because it's deprecated?

Reply:
Since `client.Options` has a way to pass information about the cache and `monitorCache` only has `Get` and `List`.

I decided to change to `client.Reader` instead.